storybookjs/storybook
# Maintenance: Add working Story Explorer config

## What I did

Some months ago, @ndelangen [reached out about support for Storybook 7](https://github.com/joshbolduc/vscode-story-explorer/issues/605#issuecomment-1352730479) in Story Explorer (my VS Code extension) for potential use in the storybook repo. SB7 support has been broadly functional for some time now, but I just finally pushed through an "official" 1.0 release and am now belatedly circling back on that initial request via this PR.

Happily, best I can tell things pretty much just work, with only one bit of important configuration required in the storybook monorepo. That's the path to the relevant Storybook config dir, since the monorepo happens to contain other Storybook configs (e.g., `test-storybooks/ember-cli/.storybook/main.js`) that are not actually part of the repo's own Storybook instance but whose existence makes the correct path ambiguous. (Hence this PR.)

<img width="1491" alt="storybook-story-explorer" src="https://github.com/storybookjs/storybook/assets/25474557/b2e53725-df6a-47be-b786-01496da3c3aa">

(A couple minor things that _don't_ work: story sorting and custom sidebar labels. This doesn't affect any other functionality, though.)

## How to test

1. Install Story Explorer in VS Code ([direct link to marketplace](https://marketplace.visualstudio.com/items?itemName=joshbolduc.story-explorer))
2. Open storybook repo in VS Code
3. A "stories" panel should appear and allow navigating stories, previewing stories, launching Storybook on-demand, etc.

## Checklist



      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

#### Maintainers

      make sure to add the `ci:merged` or `ci:daily` GH label to it.

`["cleanup", "BREAKING CHANGE", "feature request", "bug", "documentation", "maintenance", "dependencies", "other"]`
